aiaccel/hpo/samplers/nelder_mead_sampler.py

```diff
@@ -0,0 +1,184 @@
+from optuna.study import Study
+from optuna.trial import FrozenTrial, TrialState
+
+
+@dataclasses.dataclass
+class Coef:
+    r: float = 1.0
+    ic: float = -0.5
+    oc: float = 0.5
+    e: float = 2.0
+    s: float = 0.5
+
+
+class NelderMeadAlgorism:
+    def __init__(self,
+                 search_space: dict[str, list[float]],
+                 coef: Coef,
+                 seed: int | None = None,
+                 num_iterations: int = 0) -> None:
+        self._search_space = search_space
+        self.dimension: int = len(search_space)
+        self._rng: LazyRandomState = LazyRandomState(seed)
+        self.vertices: np.ndarray[float, float] = np.array([])
+        self.values: np.ndarray[float, float] = np.array([])
+        self.coef: Coef = coef
+        self.num_iterations: int = num_iterations
+        self.num_initial_create_trial: int = 0
+        self.vertex_queue: queue.Queue[float] = queue.Queue()
+
+    def __iter__(self) -> Generator[np.ndarray[float, float], None, None]:
+        # initial
+        initial_params = []
+        for _ in range(self.dimension + 1):
+            initial_param = []
+            for param_name, param_distribution in self._search_space.items():
+                search_space = {
+                    param_name: optuna.distributions.FloatDistribution(param_distribution[0], param_distribution[1])
+                    }
+                trans = _SearchSpaceTransform(search_space)
+                trans_params = self._rng.rng.uniform(trans.bounds[:, 0], trans.bounds[:, 1])
+                initial_param.append(trans.untransform(trans_params)[param_name])
+            self.num_initial_create_trial += 1
+            initial_params.append(np.array(initial_param))
+            yield np.array(initial_param), self.dimension + 1 - self.num_initial_create_trial
+        self.vertices, self.values = (
+            np.array(initial_params),
+            np.array([self.vertex_queue.get() for _ in range(self.dimension + 1)])
+            )
+
+        # nelder_mead
+        for _ in range(self.num_iterations) if self.num_iterations > 0 else itertools.count():
+            # order by
+            order = np.argsort(self.values)
+            self.vertices, self.values = self.vertices[order], self.values[order]
+            # reflect
+            self.yc = self.vertices[:-1].mean(axis=0)
+            yield (yr := self.yc + self.coef.r * (self.yc - self.vertices[-1])), 0
+            fr = self.vertex_queue.get()
+
+            if self.values[0] <= fr < self.values[-2]:
+                self.vertices[-1], self.values[-1] = yr, fr
+            elif fr < self.values[0]:
+                # expand
+                yield (ye := self.yc + self.coef.e * (self.yc - self.vertices[-1])), 0
+                fe = self.vertex_queue.get()
+                if fe < fr:
+                    self.vertices[-1], self.values[-1] = ye, fe
+                else:
+                    self.vertices[-1], self.values[-1] = yr, fr
```

Comment:
三項演算子，使えるのはここだけでしたね．すみません．
```suggestion

                self.vertices[-1], self.values[-1] = ye, fe if fe < fr else yr, fr
```

Reply:
commit suggestion押せば反映されると思いますので，活用下さい．